feat: Update Advanced Networking page (#465)
CluedIn-io/CluedIn.Documentation: docs/190-paas-operations/configuration/030-advanced-network.md

| Azure CNI | Assigns 1 IP per pod, per node based on the `maxPods` property which defaults to `50` during installation.<br>This means that if you have 5 nodes, each with a max of 50 pods, a total of 250 IP addresses will need to be used up front. | `/23`<br>510 available IP addresses | [CNI network planning](https://learn.microsoft.com/en-us/azure/aks/azure-cni-overview#plan-ip-addressing-for-your-cluster) |

{:.important}
It is best to always accomodate for expansion of your cluster, especially during AKS upgrades as additional nodes will be spun up at upgrade time. The above is CluedIn's recommendation.

## Advanced network configuration options

During the installation of CluedIn from the Azure Marketplace, you can set up advanced network configuration on the **Networking and Monitoring** tab. The default option is to create a new vNet and subnet with recommended defaults, but this can be changed to use an existing vNet.

{:.important}
If you're using an existing vNet as part of the deployment, you will need to ensure that the the AKS cluster identity has appropriate permissions to manage the subnet. This is because AKS nodes will join and leave the subnet at times.

> Out of the box, CluedIn will use a public ingress and egress load balancer to serve traffic initially regardless of vNet choice. In advanced networking setups, we recommend changing this so ingress is only accessible via a private vNet. The below will cover how to do this and as part of the CluedIn managed service, we will work with you to ensure it's done in a supported way.

Reviewer: ```suggestion
Out of the box, CluedIn will use a public ingress and egress load balancer to serve traffic initially regardless of vNet choice. In advanced networking setups, we recommend changing this so ingress is only accessible via a private vNet. The following sections will cover how to do this, and as part of the CluedIn managed service, we will work with you to ensure it's done in a supported way.
```

    In Azure, you should then see a new Load Balancer resource called `kubernetes-internal`, which will be used for ingress. The original `kubernetes` then simply becomes egress only.

## Integration options

> With Kubernetes now configured to use the internal load balancer for ingress, you will need to decide how you want to access this front end load balancer from your internal network. The two options supported are `Virtual network peering` and `Private link service with private endpoint`.

Reviewer: ```suggestion
With Kubernetes now configured to use the internal load balancer for ingress, you will need to decide how you want to access this front-end load balancer from your internal network. The two options supported are `Virtual network peering` and `Private link service with private endpoint`.
```

**Note**: If your subnets overlap in any way, the only option will be `Private link service with private endpoint`.

### Virtual network peering

> To configure virtual network peering, Microsoft have a [document](https://learn.microsoft.com/en-us/azure/virtual-network/tutorial-connect-virtual-networks-portal) explaining how to achieve this easily.
> 
> As all resources have already been created, you only need to follow the `Create virtual network peer` step in the above document.

Reviewer: ```suggestion
To configure virtual network peering, refer to [Microsoft documentation](https://learn.microsoft.com/en-us/azure/virtual-network/tutorial-connect-virtual-networks-portal). As all resources have already been created, you only need to follow the **Create virtual network peer** step in the tutorial.
```

### Private link service with private endpoint
In the event there is overlapping of subnets, this is not a problem as `private endpoints` also work here. This is slightly more complicated than virtual network peering and we'll offer a bit of guidance here.

> Microsoft have a [document](https://learn.microsoft.com/en-us/azure/private-link/create-private-link-service-portal) explaining how to achieve this. 

Reviewer: ```suggestion
Microsoft has a [document](https://learn.microsoft.com/en-us/azure/private-link/create-private-link-service-portal) explaining how to achieve this. 
```

> As AKS maintains the load balancers, and the vnet is created at deployment time (or an existing one is used), you only need to follow the steps `Create a private link service` and `Create a private endpoint` in the above document. 

Reviewer: ```suggestion
As AKS maintains the load balancers, and the vnet is created at deployment time (or an existing one is used), you only need to follow the steps **Create a private link service** and **Create a private endpoint** in the above document. 
```

For the `Private Link Service`, ensure that the `Load balancer` is set to `kubernetes-internal` and uses the frontend IP address from the private range. The Source NAT vnet and subnet should default correctly.
Ensure that Source NAT subnet is set to No and it is best to assign a Static Private IP address.

![adv-network-pls-01.png](../../assets/images/ama/howtos/adv-network-pls-01.png)

For the `Private endpoint`, ensure you select the virtual network that will be used to access CluedIn along with the subnet. It is recommended to statically allocate an IP address here as this will be used as the entry point for CluedIn and will be required as part of your DNS at a later stage.

![adv-network-pe-01.png](../../assets/images/ama/howtos/adv-network-pe-01.png)

### Next steps
The next step is to test access from the private network to CluedIn to ensure that traffic can reach the endpoint as expected.

> By default ICMP packets do not respond, but HTTPS (443/TCP) will. It may error due to certificate, but this can be corrected.

Reviewer: ```suggestion
By default, ICMP packets do not respond, but HTTPS (443/TCP) will. It may cause an error due to certificate, but this can be corrected.
```

> When you've confirmed that traffic is flowing, the final steps are to setup host name / DNS and TLS.

Reviewer: ```suggestion
When you've confirmed that traffic is flowing, the final steps are to set up host name/DNS and TLS.
```

> The below documents will help configure DNS and TLS:

Reviewer: ```suggestion
The following documents will help configure DNS and TLS:
```
